components/script/dom/webglprogram.rs

```diff
@@ -144,6 +145,9 @@ impl WebGLProgram {
         }
 
         shader_slot.set(None);
+        // TODO: This is not complete because shaders can be attached to multple programs
```

Comment:
Actually the spec says an error will be thrown if you try to attach a shader to a program if that shader is already attached to another program so a `Cell<bool>` should work fine.

> GL_INVALID_OPERATION is generated if shader is already attached to program, or if another shader object of the same type as shader is already attached to program.

https://www.khronos.org/opengles/sdk/docs/man/xhtml/glAttachShader.xml